ts/Dashboards/Components/Component.ts

```diff
@@ -1066,6 +1067,18 @@ abstract class Component {
         return json;
     }
 
+    /**
+     * Get the component's options.
+     * @returns
+     * The JSON of component's options.
+     *
+     * @internal
+     *
+     */
+    public getOptions(): Partial<Component.ComponentOptions> {
+        return diffObjects(this.options, Component.defaultOptions);
+    }
```

Comment:
Since all of the inheriting classes are implementing their own version of this method, I'd use the abstract method here, and force those classes to implement their version.

```suggestion
    /**
     * Get the component's options.
     * @returns
     * The JSON of component's options.
     *
     * @internal
     *
     */
    public abstract getOptions(): Partial<Component.ComponentOptions>;
```

Reply:
It would require refactoring KPI parts and ThresholdComponent, so I don't think that is valuable at this moment.

Reply:
Your call